### PR TITLE
feat: auto connect & fix wallets not showing up

### LINKED ISSUE #228: Add auto connect like in StarkFighter
**Is your feature request related to a problem? Please describe.**
Add auto-connect like in the starkfighter app.

**Describe the solution you'd like**
if I reload I stay connected, If I quit the app and come back tomorrow I also stay connected ...

### PR DESCRIPTION
**Pull request by irisdv**
close #228 & #248 

this PR also bumps starknet-react to version 0.17.0

### CODE REVIEW

File: components/UI/navbar.tsx [hunk 1]
```diff
@@ -30,8 +30,14 @@ const Navbar: FunctionComponent = () => {
     disconnect();
     setIsConnected(false);
     setIsWrongNetwork(false);
+    setHasWallet(false);
   }
 
+  useEffect(() => {
+    const interval = setInterval(refresh, 5000);
+    return () => clearInterval(interval);
+  }, [refresh]);
```
Comment (fricoben) [inline components/UI/navbar.tsx]: Là tu est constamment en train de refresh ?

File: components/UI/navbar.tsx [hunk 2]
```diff
@@ -231,10 +241,12 @@ const Navbar: FunctionComponent = () => {
...
       />
-      <Wallets
-        closeWallet={() => setHasWallet(false)}
-        hasWallet={Boolean(hasWallet && !isWrongNetwork)}
-      />
+      {hasWallet ? (
+        <Wallets
+          closeWallet={() => setHasWallet(false)}
+          hasWallet={Boolean(hasWallet && !isWrongNetwork)}
+        />
```
Comment (fricoben) [inline components/UI/navbar.tsx]: Pas sur que ce soit nécessaire. 

La props has wallet sert justement à éviter ce ternary

File: components/domains/register.tsx [hunk 3]
```diff
@@ -4,14 +4,19 @@ import { FunctionComponent, useEffect, useState } from "react";
 import Button from "../UI/button";
 import styles from "../../styles/Home.module.css";
 import { useEtherContract, usePricingContract } from "../../hooks/contracts";
-import { Call, useAccount, useStarknetCall } from "@starknet-react/core";
-import { useStarknetExecute } from "@starknet-react/core";
+import {
+  Call,
```
Comment (fricoben) [inline components/domains/register.tsx:8]: `Call` pas utilisé

File: hooks/naming.ts [hunk 4]
```diff
@@ -110,11 +111,12 @@ export function useExpiryFromDomain(domain: string): ExpiryData {
...
-    method: "domain_to_expiry",
+  const { data, error } = useContractRead({
+    address: contract?.address as string,
+    abi: contract?.abi as Abi,
+    functionName: "domain_to_expiry",
     args: [encoded],
   });
 
-  return { expiry: data as any, error };
+  return { expiry: data as any, error: error as string };
```
Comment (fricoben) [inline hooks/naming.ts:121]: Pourquoi dois-tu forcer le type ici ? error n'est pas déjà du string ?

**Review by fricoben — CHANGES_REQUESTED** [review]
Gud PR ! Some comments and questions

File: components/UI/navbar.tsx (hunk 1 above)
Comment (irisdv) [reply components/UI/navbar.tsx]: oui en effet, je peux stoper le refresh quand le user est connecté sinon

File: hooks/naming.ts (hunk 4 above)
Comment (irisdv) [reply hooks/naming.ts:121]: Dans starknet-react `error` a un type `unknown` et dans `ExpiryData` error est `string | undefined`

File: components/domains/register.tsx (hunk 3 above)
Comment (irisdv) [reply components/domains/register.tsx:8]: A la ligne 34 dans `const [callData, setCallData] = useState<Call[]>([])`; non ?

File: components/UI/navbar.tsx (hunk 2 above)
Comment (irisdv) [reply components/UI/navbar.tsx]: je l'ai enlevé

File: components/UI/navbar.tsx [hunk 5]
```diff
@@ -13,7 +13,7 @@ import { useDisplayName } from "../../hooks/displayName.tsx";
 
 const Navbar: FunctionComponent = () => {
   const [nav, setNav] = useState<boolean>(false);
-  const [hasWallet, setHasWallet] = useState<boolean>(true);
+  const [hasWallet, setHasWallet] = useState<boolean>(false);
```
Comment (fricoben) [inline components/UI/navbar.tsx:16]: Is it necessary ? As I tested, the auto-connect works well 🔥

But when you are new to the app, you have to click on connect on the top right button in order to open this. It's a better UX to directly open it only if the user never connected imo. 

Wdyt ?

**Review by fricoben — CHANGES_REQUESTED** [review]
Just a last question concerning first time users and the way to handle them.

File: components/UI/navbar.tsx (hunk 5 above)
Comment (irisdv) [reply components/UI/navbar.tsx:16]: Me, I prefer when it doesn't popup even when I'm new, but that's my personal point of view :)

If it is set to true at first and the user has already connected once then the popup is shown for less than a second before it disappears. What I could do is to keep it to `false` whenever a user connects, and check local storage: 
- if there is a value for `lastUsedConnector` then `hasWallet` stays to `false`, 
- if there is no value, set `hasWallet` to `true` and show the popup wallet.

Comment (fricoben) [reply components/UI/navbar.tsx:16]: It's a bit complicated here.

Or maybe we could directly connect the user to the first wallet on the list if he is not connected? This means it'll directly open his Argent/Braavos app and ask him to approve the connection.

Wdyt ?

**Review by fricoben — APPROVED** [review]
LGTM

**Pull request merged**